src/System.Private.CoreLib/shared/System/Runtime/MemoryFailPoint.Unix.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime
+{
+    public sealed partial class MemoryFailPoint
+    {
+        private static ulong GetTopOfMemory()
+        {
+            return ulong.MaxValue;
```

Comment:
This is not correct for 32-bit platforms.

Reply:
It is not correct for any CoreCLR platform in fact. It is deliberately set to a large enough value that it won't accidentally trigger an error. Unfortunately there's no `UIntPtr.MaxValue`, which would be more appropriate.

The original PAL implementation did this:
```c
#ifdef VM_MAXUSER_ADDRESS
    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) VM_MAXUSER_ADDRESS;
#elif defined(__linux__)
    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) (1ull << 47);
#elif defined(USERLIMIT)
    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) USERLIMIT;
#elif defined(_WIN64)
#if defined(USRSTACK64)
    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) USRSTACK64;
#else // !USRSTACK64
#error How come USRSTACK64 is not defined for 64bit?
#endif // USRSTACK64
#elif defined(USRSTACK)
    lpSystemInfo->lpMaximumApplicationAddress = (PVOID) USRSTACK;
#else
#error The maximum application address is not known on this platform.
#endif
```

It was only marginally more correct and it didn't care for 32-bit Linux platforms either.